statsmodels/tsa/seasonal.py

```diff
@@ -30,7 +30,7 @@ def seasonal_decompose(x, model="additive", filt=None, freq=None):
         The filter coefficients for filtering out the seasonal component.
         The default is a symmetric moving average.
     freq : int, optional
-        Frequency of the series. Must be used if x is not a pandas
+        Frequency of the series. Overrides default periodicity of x if x is a pandas
```

Comment:
The old sentence that `freq` has to be provided if x is not pandas object needs to stay there. The new behavior won't change that.